src/UI/Art/SpriteArtist/SpriteUIUtils.java

```diff
@@ -108,6 +117,51 @@ public static BufferedImage makeTextMenu(Color bg, Color frame, Color focus, Arr
     return menuImage;
   }
 
+  /**
+   * Returns an image with the input string printed within the specified width, in normal text.
+   * @param reqWidth: Actual UI size in pixels that you want to fit the text into.
+   */
+  public static BufferedImage paintTextNormalized(String prose, int reqWidth)
+  {
+    // Figure out how big our text is.
+    int drawScale = SpriteOptions.getDrawScale();
+    int characterWidth = SpriteLibrary.getLettersUppercase().getFrame(0).getWidth() * drawScale;
+    int characterHeight = SpriteLibrary.getLettersUppercase().getFrame(0).getHeight() * drawScale;
+
+    ArrayList<String> lines = new ArrayList<String>();
+    // Unload our prose into the lines it already has
+    lines.addAll(Arrays.asList(prose.split("\\R"))); // \R matches all newline formats, yay convenience
+
+    for( int i = 0; i < lines.size(); ++i ) // basic for, since we care about indices
+    {
+      String line = lines.get(i);
+      if( line.length() * characterWidth <= reqWidth ) // if the line's short enough already, don't split it further
+        continue;
+
+      // TODO: check for word boundaries
+      lines.remove(i);
+      lines.add(i, line.substring(reqWidth / characterWidth)); // put in the second half
+      lines.add(i, line.substring(0, reqWidth / characterWidth)); // and then the first half behind it
+    }
+
+    int totalTextHeight = ((lines.isEmpty()) ? 0 : getMenuTextHeightPx(lines, characterHeight));
+    // Build our image.
+    BufferedImage menuImage = null;
+    if( reqWidth == 0 || totalTextHeight == 0 )
```

Comment:
if `reqWidth == 0`, you would have hit an infinite loop condition where you are breaking lines apart. If you want to validate inputs, do it at the beginning of the function.